fern/apis/dynamic-snippets/definition/snippets.yml

```diff
@@ -0,0 +1,47 @@
+imports:
+  endpoints: ./endpoints.yml
+
+types:
+  AuthValues:
+    union:
+      basic: BasicAuthValues
+      bearer: BearerAuthValues
+      header: HeaderAuthValues
+      oauth: OAuthValues
+
+  BasicAuthValues:
+    properties:
+      username: string
+      password: string
+
+  BearerAuthValues:
+    properties:
+      token: string
+
+  HeaderAuthValues:
+    properties:
+      value: unknown
+
+  OAuthValues:
+    properties:
+      clientId: string
+      clientSecret: string
+
+  Values:
+    docs: |
+      Snippet values are represented as arbitrary key, value
+      pairs (i.e. JSON objects). The keys are expected to be
+      in the parameter's wire representation. For path parameters,
+      the name will match the parameter name.
+    type: map<string, unknown>
+
+  SnippetRequest:
```

Comment:
is it useful to encode a `SnippetResponse` (will make it easier to not break that API either)

Reply:
There's a lot we can add for responses to include in snippet generation (e.g. whether the endpoint responds with pagination, streaming, etc). 

The caller shouldn't need to specify any additional values in the `SnippetRequest` though - I imagine this would be an additional `response` property on the `Endpoint` type like so:

```yaml
types:
  Endpoint:
    properties:
      auth: optional<EndpointAuth>
      declaration: declaration.Declaration
      request: Request
      response: Response

  Request:
    docs: |
      Reperesents the request parameters required to call a specific endpoiont.
    union:
      body: BodyRequest
      inlined: InlinedRequest

  Response:
    docs: |
      Reperesents the response returned from the endpoint.
    union:
      unary: UnaryResponse
      fileDownload: FileDownloadResponse
      pagination: PaginationResponse
      streaming: StreamingResponse
```

With backwards compatibility in mind, I'm happy to start with just the `UnaryResponse` types as an empty object (it might someday have `json`, `text`, `bytes`, and so on). When this `UnaryResponse` with no properties is specified, the snippet will just be the traditional endpoint call like so:

```go
import (
  context "context"
  acme "github.com/buildwithfern/acme-go"
  client "github.com/buildwithfern/acme-go/client"
)

client := client.NewClient()
client.User.Create(
  context.TODO(),
  &acme.CreateUserRequest{
    Username: "john.doe",
    Status:   acme.StatusDeactivated,
  },
)
```

Then we can keep `response` typed as a required property, then roll out `FileDownload`, `PaginationResponse` and `StreamingResponse` later on in safe way.

--

Thoughts?

Reply:
Discussed offline - the above looks good (and is included here), but we should still split out a separate `EndpointSnippetResponse` that contains a property for the snippet string. That way it's easy for us to support additional snippet variants later on.